### PR TITLE
fix some tests in prep for new initialization, 

### PR DESCRIPTION
**Pull request by dlwh**
which caught some problems

tagging @rjpower . Turns out we need to use approx=True for gelu in JAX to really match Gemma

### CODE REVIEW

**Comment by dlwh** [comment]
(also i know the hasattr is gross, I'll clean it up)

File: src/levanter/models/llama.py [hunk 1]
```diff
@@ -257,6 +257,13 @@ def init(config: LlamaConfig, *, key) -> "LlamaAttention":
         )
         return LlamaAttention(config, q_proj, k_proj, v_proj, o_proj)
 
+    def _rope_scale_factor(self) -> float:
```
Comment (rjpower) [inline src/levanter/models/llama.py:260]: Feel free to file an issue and I can look at normalizing how Gemma handles this. I can't remember if HFGemma doesn't use it at all or if it was doing something different and weird.

**Review by rjpower — APPROVED** [review]
Nice catch! Here I went through all those hoops to deal with the weird HF activation handling and I managed to replicate the same issue...

File: src/levanter/models/llama.py (hunk 1 above)
Comment (dlwh) [reply src/levanter/models/llama.py:260]: HfGemma doesn't use it at all, which seems like an oversight more than anything. (Llama 2 also doesn't use it by default, but you can turn it on and do some tuning to get the length extrapolation, I think)

**Comment by dlwh** [comment]
yeah, these things are kind of finicky. It's so easy to have something that seems right but it's wrong in some tiny detail like this. It doesn't help that gelu and gelu_new are completely confusing

**Pull request merged**